### PR TITLE
Odb store routing layer range

### LINKED ISSUE #5594: odb: store routing layer range in db instead of grt
### Description

This is required to avoid problems with RUDY heatmap when ORFS scripts initiate grt but not fastroute. Further details in PR https://github.com/The-OpenROAD-Project/OpenROAD/pull/5578/ and issue https://github.com/The-OpenROAD-Project/OpenROAD/issues/5574

The layer range data is stored inside GRT data structures. In cases where RUDY need to access this data before reading a script that defines the layer range, we need the range data inside odb.

### Suggested Solution

Create a structure that stores the layer ranges used by GRT (for signal nets and for clock nets_) in odb. odb::dbBlock seems to be a good place for it. Update GRT code to access the layer range data from the odb instead of coming directly from the Tcl script.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by LucasYuki**
Closes #5594.

Changes:
- Layer ranges moved from GRT to odb::dbBlock. (Implements #5594)
- TCL commands related to layer ranges moved from GRT to ODB.
- Documentation about manually adding new fields in DB Object updated.

### CODE REVIEW

File: src/odb/doc/OpenDB-AddFieldsInDbObjects.md [hunk 1]
```diff
@@ -1,15 +1,113 @@
...
-| 9 | Add new APIs in include/db.h                           | db.h              | `class dbTechLayer`                                                                |
+*Modifications in these steps need to be done inside user regions, delimited like the below, to avoid being rewritten by the code generator
+```cpp
+// User Code Begin <something>
+
+...
+
+// User Code End <something>
+```
+<table>
```
Comment (maliberty) [inline src/odb/doc/OpenDB-AddFieldsInDbObjects.md:13]: Markdown can generate tables - is there a reason to use raw html?

**Comment by maliberty** [comment]
The code looks good.

File: src/odb/doc/OpenDB-AddFieldsInDbObjects.md (hunk 1 above)
Comment (LucasYuki) [reply src/odb/doc/OpenDB-AddFieldsInDbObjects.md:13]: I couldn't put a code block with multiple lines with Markdown, so I used html.

**Comment by eder-matheus** [comment]
@LucasYuki One of your commits ([68aa6f0](https://github.com/The-OpenROAD-Project/OpenROAD/pull/5835/commits/68aa6f07e299c4de0def3cef15d0c164f5352b62)) doesn't have the signoff, and it can be painful to fix.

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
I've waived the DCO on this one since but please remember to use `commit -s` in the future.  Fwiw I have in my .gitconfig
```
[alias]
        ci = commit -s
```
so I do `git ci ...` and don't forget.

**Pull request merged**